igor-web/src/test/groovy/com/netflix/spinnaker/igor/wercker/WerckerBuildMonitorSchedulingSpec.groovy

```diff
@@ -0,0 +1,137 @@
+        werckerService.getRunsSince(_) >> [:]
+        def cfg = new IgorConfigurationProperties()
+        cfg.spinnaker.build.pollInterval = 1
+        monitor = new WerckerBuildMonitor(
+                cfg,
+                new NoopRegistry(),
+                Optional.empty(),
+                Optional.empty(),
+                cache,
+                buildMasters,
+                true,
+                Optional.empty(),
+                Optional.empty(),
+                new WerckerProperties()
+                )
+        monitor.worker = scheduler.createWorker()
+        when:
+        monitor.onApplicationEvent(Mock(RemoteStatusChangedEvent))
+        scheduler.advanceTimeBy(1L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then: 'initial poll'
+        1 * buildMasters.filteredMap(BuildServiceProvider.WERCKER) >> [MASTER: werckerService]
+        1 * buildMasters.map >> [MASTER: werckerService]
+        1 * werckerService.getRunsSince(_) >> [:]
+
+        when:
+        scheduler.advanceTimeBy(998L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then:
+        0 * buildMasters.map >> [MASTER: werckerService]
+        0 * werckerService.getRunsSince(_) >> [:]
+
+        when: 'poll at 1 second'
+        scheduler.advanceTimeBy(2L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then:
+        1 * buildMasters.filteredMap(BuildServiceProvider.WERCKER) >> [MASTER: werckerService]
+        1 * buildMasters.map >> [MASTER: werckerService]
+        1 * werckerService.getRunsSince(_) >> [:]
+
+        when: 'poll at 2 and 3 second'
+        scheduler.advanceTimeBy(4000L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then:
+        4 * buildMasters.filteredMap(BuildServiceProvider.WERCKER) >> [MASTER: werckerService]
+        4 * buildMasters.map >> [MASTER: werckerService]
+        4 * werckerService.getRunsSince(_) >> [:]
+
+        cleanup:
+        monitor.stop()
+    }
+
+    void 'scheduler can be turned off'() {
+        given:
+        cache.getJobNames(MASTER) >> []
+        BuildMasters buildMasters = Mock(BuildMasters)
+        def cfg = new IgorConfigurationProperties()
+        cfg.spinnaker.build.pollInterval = 1
+        monitor = new WerckerBuildMonitor(
+                cfg,
+                new NoopRegistry(),
+                Optional.empty(),
+                Optional.empty(),
+                cache,
+                buildMasters,
+                false,
+                Optional.empty(),
+                Optional.empty(),
+                new WerckerProperties()
+                )
+        monitor.worker = scheduler.createWorker()
+
+        when:
+        monitor.onApplicationEvent(Mock(RemoteStatusChangedEvent))
+        scheduler.advanceTimeBy(1L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then: 'initial poll'
+        0 * buildMasters.filteredMap
+
+        when:
+        scheduler.advanceTimeBy(998L, TimeUnit.SECONDS.MILLISECONDS)
+
+        then:
+        0 * buildMasters.filteredMap
+
+        when: 'poll at 1 second'
```

Comment:
Minor, but maybe should change this to "no poll at 1 second" or something similar, as we're expecting the absence of polling.

Reply:
fixed.